### PR TITLE
Using custom yaml loader for topological map file

### LINKED ISSUE #185: [FEAT]: Yaml load with extended safe loader for topological map files
### Problem:

The current map managers loads the topological map file using the PyYaml safe load option directly [here on manager2.py file](https://github.com/LCAS/topological_navigation/blob/0217c0eec73bfa8b16e8360c87b48dd811d6e6cf/topological_navigation/topological_navigation/manager2.py#L205): 

Considering a map file that is exported using JSON stringily for large farms, the 'stringification' strips all the leading zeros so a pose orientation like the following : 

```js
const orientation = { x : 0.0, y : 0.0, z : 0.0, w : 1.0 } 
```

would become: 

```yaml
orientation: 
      x : 0
      y : 0
      z : 0
      w : 1
```

When the yaml file is loaded in Python, the x, y, z, w keys are treated as integer type instead of float types. This integer type fails an assertion in [geometry_msgs/Point](https://github.com/ros2/common_interfaces/blob/0163014f208bbe840a5c2d23d98dc27fc0d02b2c/geometry_msgs/msg/Point.msg#L1) and [geometry_msgs/Vector3](https://github.com/ros2/common_interfaces/blob/0163014f208bbe840a5c2d23d98dc27fc0d02b2c/geometry_msgs/msg/Vector3.msg#L1): 

The behaviour is reproducible if any of the nodes' pose or map global transformation is set without leading zeroes. 

### Solution: 

PyYaml allows to extend the behaviour of SafeLoad by extending the class and passing it as a Loader argument. Within that extension, one can change the 'x', 'y', 'z', and 'w' keys to float type. [Something like the first response here](https://stackoverflow.com/questions/74334528/pyyaml-load-version-number-float-with-trailing-zero-as-string). 

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by arsh09**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR fixes #185 with custom yaml loader. 

Additionally, the same custom loader class can be extended to check for missing fields or check node attributes before loading the map instead. See [here](https://github.com/arsh09/topological_navigation/blob/02f0d9d605d6a430c8cec08e921556ff00a76994/topological_navigation/topological_navigation/manager2.py#L41)

### CODE REVIEW

File: topological_navigation/test/test_navigationcore.py [hunk 1]
```diff
@@ -94,8 +94,8 @@ def __init__(self, name='test_node'):
 
     def initialize(self):
         self.topmap_sub = self.create_subscription(String, '/topological_map_2', self.topmap_sub_callback, 1)
-        self.closest_node_sub = self.create_subscription(String, 'closest_node', self.closest_node_callback, 1)
-        self.current_node_sub = self.create_subscription(String, 'current_node', self.current_node_callback, 1)
+        self.closest_node_sub = self.create_subscription(String, '/closest_node', self.closest_node_callback, 1)
+        self.current_node_sub = self.create_subscription(String, '/current_node', self.current_node_callback, 1)
```
Comment (marc-hanheide) [inline topological_navigation/test/test_navigationcore.py]: I don't think we want an absolute topic URI here. This prevents it from starting in namespace which need

File: topological_navigation/test/test_navigationcore.py [hunk 2]
```diff
@@ -94,8 +94,8 @@ def __init__(self, name='test_node'):
 
     def initialize(self):
         self.topmap_sub = self.create_subscription(String, '/topological_map_2', self.topmap_sub_callback, 1)
-        self.closest_node_sub = self.create_subscription(String, 'closest_node', self.closest_node_callback, 1)
-        self.current_node_sub = self.create_subscription(String, 'current_node', self.current_node_callback, 1)
+        self.closest_node_sub = self.create_subscription(String, '/closest_node', self.closest_node_callback, 1)
```
Comment (marc-hanheide) [inline topological_navigation/test/test_navigationcore.py]: see below, no absolute topic URI please change

File: topological_navigation/topological_navigation/manager2.py [hunk 3]
```diff
@@ -151,7 +168,7 @@ def init_map(self, name="new_map", metric_map="map_2d", pointset="new_map", tran
         self.loaded = False
         self.load = load
         if self.load:
-            self.load_map(self.filename)
+            self.load_map(self.filename)
```
Comment (marc-hanheide) [inline topological_navigation/topological_navigation/manager2.py:171]: spurious space

File: topological_navigation/topological_navigation/manager2.py [hunk 4]
```diff
@@ -38,6 +38,23 @@ def ignore_aliases(self, data):
...
+
+        # this can be extended to test the validity of the tmap2 
+        # as well at load time (or add missing keys)
+        for key in ['x', 'y', 'z', 'w']:
+            if key in mapping and isinstance(mapping[key], int):
+                mapping[key] = float(mapping[key])
+        
+        return mapping
+
+
```
Comment (marc-hanheide) [inline topological_navigation/topological_navigation/manager2.py:57]: looks good

**Review by marc-hanheide — CHANGES_REQUESTED** [review]
the absolute topic names are a problem

File: topological_navigation/topological_navigation/manager2.py (hunk 3 above)
Comment (arsh09) [reply topological_navigation/topological_navigation/manager2.py:171]: Sorry for this.

File: topological_navigation/test/test_navigationcore.py (hunk 2 above)
Comment (arsh09) [reply topological_navigation/test/test_navigationcore.py]: Changed it back to relative topic URIs.

**Review by ibrahimhroob — APPROVED** [review]
All looks fine

**Review by ibrahimhroob — APPROVED** [review]

**Review by marc-hanheide — APPROVED** [review]

**Pull request merged**